File: Problems/0283.Move-Zeroes/MoveZeroes_test.py

from MoveZeroes import Solution
import pytest
from copy import deepcopy

SOL = Solution()

TEST_SUITS =  [
    ([0,1,0,3,12], [1,3,12,0,0]),
    ([2,1], [2,1]),
    ([4,2,4,0,0,3,0,5,1,0], [4,2,4,3,5,1,0,0,0,0]),
]

@pytest.mark.parametrize(
    "nums, ans", 
    deepcopy(TEST_SUITS)
)
def test(nums, ans):
    SOL.moveZeroes(nums)
    assert nums == ans
    

File: Problems/0283.Move-Zeroes/MoveZeroes_test.go
package leetcode

import (
	"reflect"
	"testing"
)

func Test(t *testing.T) {
	testSuits := []struct {
		name string
		nums []int
		ans  []int
	}{
		{"Example 1", []int{0, 1, 0, 3, 12}, []int{1, 3, 12, 0, 0}},
		{"Example 2", []int{2, 1}, []int{2, 1}},
		{"Example 3", []int{4, 2, 4, 0, 0, 3, 0, 5, 1, 0}, []int{4, 2, 4, 3, 5, 1, 0, 0, 0, 0}},
	}

	for _, test := range testSuits {
		moveZeroes(test.nums)

		if !reflect.DeepEqual(test.nums, test.ans) {
			t.Errorf("[%v] expect '%v', got '%v' ", test.name, test.ans, test.nums)
		}
	}
}
